archeryutils/handicaps/handicap_scheme_aa.py

```diff
@@ -0,0 +1,296 @@
+"""Code for handicap calculations using Archery Australia handicap schemes.
+
+Extended Summary
+----------------
+Code to calculate information using a number of schemes:
+
+- Original Archery Australia (J Park)
+- Updated Archery Australia (J Park)
+
+Routine Listings
+----------------
+- HandicapAA Class
+- HandicapAA2 Class
+
+References
+----------
+- Modelling archers’ scores at different distances to quantify
+  score loss due to equipment selection and technique errors
+  Park, J (2014)
+  https://doi.org/10.1177%2F1754337114539308
+
+"""
+
+import numpy as np
+
+from .handicap_scheme import FloatArray, HandicapScheme
+
+
+class HandicapAA(HandicapScheme):
+    """
+    Class to represent the original Archery Australia scheme by J. Park.
+
+    Parameters
+    ----------
+    ang_0 : float
+        datum angle used in scheme, default = 1.0e-3,
+    k0 : float
+        handicap offset to set scratch point, default= 2.37,
+    ks : float
+        change with each handicap step as a geometric progression, default= 0.027,
+    kd : float
+        distance scaling factor, default= 0.004,
+
+    Attributes
+    ----------
+    name : str="AA"
+        The name of the handicap scheme.
+    arw_d_out : float
+        diameter of an outdoor arrow: 5.0e-3 [metres]
+    arw_d_in: float
+        diameter of an indoor arrow 9.3e-3 [metres]
+
+    Methods
+    -------
+    sigma_t(handicap, dist)
+        Calculate angular deviation for given handicap and distance.
+    sigma_r(handicap, dist)
+        Calculate radial deviation for a given handicap and distance.
+    arrow_score(target, handicap, arw_d=None)
+        Calculate the average arrow score for a given target and handicap.
+    score_for_passes(rnd, handicap, arw_d=None, rounded_score=True)
+        Calculate the expected score for all passes in a round for a given handicap.
+    score_for_round(rnd, handicap, arw_d=None, rounded_score=True)
+        Calculate the expected score for a round for a given handicap.
+    handicap_from_score(score,rnd, arw_d=None, int_prec=False)
+        Calculate the handicap for a given score on a given round.
+
+    Warnings
+    --------
+    Using non-default values for the kwargs may produce results that
+    do not match the official values for this scheme.
+    Only change if you are experimenting and know what you are doing!
+
+    See Also
+    --------
+    HandicapScheme : The base class for a handicap scheme from which this is subclassed.
+
+    """
+
+    def __init__(
+        self,
+        ang_0: float = 1.0e-3,
+        k0: float = 2.37,
+        ks: float = 0.027,
+        kd: float = 0.004,
+    ):
+        self.params = {
+            "ang_0": ang_0,  # Baseline angle used for group size 1.0 [millirad].
```

Comment:
I can't see why the parameters are stored within the dict `self.params` rather than just being regular variables. Is it just to ensure all subclasses have the same contents?

Reply:
Yeah, I also don't yet understand why the indirection here. A quick search on the project and I think `.params` is only accessed in the class specific `sigma_t` methods, which means that every method can reasonably know about what parameters it should be able to access.

Reply:
I can understand it if the user wants to know what their current setup is, as calling `my_scheme.params` is much easier than parsing the results of `vars(my_scheme)` and trying to figure out which variables relate to the scheme. I'm not sure how likely this is to happen though, and removing the indirection would make the code in `sigma_t` a little cleaner. It's really not a huge deal either way though.

Reply:
Could always generate that dict of params as a property if you want it easily exported. It just feels like an unusual pattern to me to pass them as individual instance variables but then access them bundled through the dict. If I was looking to to contribute to the project with a new handicap scheme that would certainly have be scratching my head a little.

Reply:
Passing in as separate variables means you could change/override just one value without having to also pass all of the others in, and if you did want to pass multiple you could do it with a `kwargs` dict.

e.g. I can see an instance where people want to amend the 'zero handicap' position without changing the general scaling - this was indeed something I played around with whilst developing the scheme trying to get a scratch point that was at 1400 on a 1440 or 700 on a 720.

I don't think this one needs overthinking as 99.9% of users are unlikely to ever touch things at this depth in the code. Having it as a dict may come in useful when writing a `__repr__` method...

Reply:
What do you think about #79 which would neaten this up?

Reply:
As detailed over in #79 that has been abandoned and sticking to form as discussed above.